adapter/internal/oasparser/envoyconf/routes_with_clusters.go

```diff
@@ -1353,12 +1360,16 @@ func generatePathRegexSegment(resourcePath string) string {
 	return resourceRegex
 }
 
+func generatePathRegexSegmentForSubstitutionString(resourcePath string) string {
```

Comment:
```suggestion
// generatePathRegexSegmentForSubstitutionString generates the regex for substitution string. Here the dots should not be transformed to regex compared to generating router matching regex but the path params are. 
func generatePathRegexSegmentForSubstitutionString(resourcePath string) string {
```

Reply:
Added the function description with: https://github.com/wso2/product-microgateway/pull/3415/commits/084b4812aa3792bb909882f644265650c91f4358